### PR TITLE
add bogo_sort.c

### PR DESCRIPTION
**Pull request by riya-17**
#3183 

Add bogo sort in c language

### CODE REVIEW

**Comment by riya-17** [comment]
@YashovardhanSiramdas i gave gone through it, and made changes. it is still not working.

**Comment by ritwik12** [comment]
@riya-17 That's ohk.It's not your fault.
I can see that Travis is not working properly, you can see that every PR is getting failed by Travis.
Travis.yml needs changes to be done.

**Comment by YashovardhanSiramdas** [comment]
@riya-17 I wasn't talking about Travis Check. Travis CI is still not deployed for C programs. I was pointing to style Guide violations.

File: code/sorting/src/bogo_sort/bogo_sort.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/*
+* bogo_sort.c
+* created by Riya
+*/
+
+#include <stdio.h>    
+#include <stdlib.h>    
+
+int main()
```
Comment (YashovardhanSiramdas) [inline code/sorting/src/bogo_sort/bogo_sort.c]: `int
main()`

Comment (YashovardhanSiramdas) [inline code/sorting/src/bogo_sort/bogo_sort.c]: Function return type and definition should be in separate lines
`int'
'main()'

File: code/sorting/src/bogo_sort/bogo_sort.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+* bogo_sort.c
+* created by Riya
+*/
+
+#include <stdio.h>    
+#include <stdlib.h>    
+
+int main()    
+{        
+    int num[10]={1,4,7,5,9,2,6,3,8,0};
```
Comment (YashovardhanSiramdas) [inline code/sorting/src/bogo_sort/bogo_sort.c]: Necessary spacing is required
`int num[10] = {1, 4, 7, 5, 9, 2, 6, 3, 8, 0};`

File: code/sorting/src/bogo_sort/bogo_sort.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    for (i = 0; i<10 ;i++) {            
+        printf("%d\n", num[i]);        
+    } printf("\n");    
+}         
+int is_sorted(int *a, int n)    
+{        
+    while ( --n >= 1 ) {        
+        if ( a[n] < a[n-1] ) 
+            return (0);    
+    } return (1);
```
Comment (YashovardhanSiramdas) [inline code/sorting/src/bogo_sort/bogo_sort.c]: `return` statement should be in a new line.

File: code/sorting/src/bogo_sort/bogo_sort.c [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#include <stdio.h>    
+#include <stdlib.h>    
+
+int main()    
+{        
+    int num[10]={1,4,7,5,9,2,6,3,8,0};        
+    int i;            
+    bogosort(num, 10);        
+    printf("The array after sorting is:");        
+    for (i = 0; i<10 ;i++) {
```
Comment (YashovardhanSiramdas) [inline code/sorting/src/bogo_sort/bogo_sort.c]: Give necessary spaces

**Comment by ritwik12** [comment]
@YashovardhanSiramdas Sorry I was talking about build fail.

**Comment by riya-17** [comment]
@YashovardhanSiramdas @ritwik12 Thanku both

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Comment by riya-17** [comment]
@AdiChat  Thanku. Yes I am enjoying it :)

**Pull request merged**